indico_hub/server.py

```diff
@@ -42,4 +48,113 @@ def _openapi(test, as_json, host, port):
             print(spec.to_yaml())
 
 
+
 # TODO: Implement the API endpoints here
+
+
+@api.route("/api/instance", methods= ["POST"])
+def register():
+    """
+    mock function for registering an instance to the database
+    ---
+    parameters: 
+        Instance.get_json()
+    
+    
+
+    responses:
+          201:
+            description: instance registered 
+          404:
+            description: instance is already registered
+    """
+    
+    print("validating params...")
+    
+    errors = ValidationSchema().validate(request.form)
```

Comment:
Please use webargs for this. Also, the API uses JSON data, not form data.

Check [here](https://github.com/indico/openreferee/blob/master/openreferee_server/server.py#L89) now how this can be done.

Reply:
Hi, I have been trying to solve this in the last two hours.
So I am having a problem when I use @use_kwargs that the server returns a 422 error code for me. I have been looking online for a long time but couldn't find a post that has a similar issue
Here is a very simple code I have written but it is still returning 422 err:
`@api.route("/api/testingwebargs", methods=("POST", "PATCH", "PUT"))
@use_kwargs(UpdateInstance, location="json")
def testWebargs(enabled, url, contact, email, organization):
    print(f"{enabled}, {url}, {contact}, {email}, {organization}")
    return "hello"`

Reply:
Are you sending standard form data instead of JSON by any chance?

Reply:
Assuming you use tests.py then that is indeed the problem: you need to use `json=payload` for requests to send the data as JSON.

Reply:
No I have changed my tests since the beginning to use requests.post(..., json= payload)

Reply:
It works fine when I use it with an actual flask app
but it is causing issues when combined with a blueprint which is odd

Reply:
the 422 should include details on how the validation failed. check the `resp.json()` of your requests call

Reply:
thnx, found out the err, I was checking the response with if stmnt before printing it.
Anyways it was confusing =)

Reply:
Solved and commited !